packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx

```diff
@@ -1,24 +1,286 @@
+
+        renderWithProviders(<UserCompletionModal />, {
+            user: {
+                isSetupComplete: false,
+                organizationName: '',
+                email: 'demo@lightdash.com',
+            },
+            health: {
+                mode: LightdashMode.DEFAULT,
+            },
+        });
+
+        // submit button should be disabled until organization name and role are filled
+        const submitButton = await screen.findByRole('button', {
+            name: 'Next',
+        });
+        expect(submitButton).toBeDisabled();
+
+        // fill in organization name
+        const nameInput = await screen.findByPlaceholderText(
+            'Enter company name',
+        );
+        expect(nameInput).toBeInTheDocument();
+        await user.type(nameInput, 'test organization');
+
+        // select role
+        const roleSelect = await screen.findByPlaceholderText(
+            'Select your role',
+        );
+        expect(roleSelect).toBeInTheDocument();
+        await user.click(roleSelect);
+        const roleOption = await screen.findByText('Software Engineer');
+        await user.click(roleOption);
+
+        // uncheck email domain checkbox (checked by default)
+        const emailDomainCheckbox = await screen.findByRole('checkbox', {
+            name: `Allow users with @lightdash.com to join the organization as a viewer`,
+        });
+        expect(emailDomainCheckbox).toBeInTheDocument();
+        expect(emailDomainCheckbox).toBeChecked();
+        await user.click(emailDomainCheckbox);
+
+        // uncheck subscription checkbox (unchecked by default)
+        const subscriptionCheckbox = await screen.findByRole('checkbox', {
+            name: `Keep me updated on new Lightdash features`,
+        });
+        expect(subscriptionCheckbox).toBeInTheDocument();
+        expect(subscriptionCheckbox).toBeChecked();
+        await user.click(subscriptionCheckbox);
+
+        // check tracking checkbox (unchecked by default)
+        const trackingCheckbox = await screen.findByRole('checkbox', {
+            name: `Anonymize my usage data`,
+        });
+        expect(trackingCheckbox).not.toBeChecked();
+        expect(trackingCheckbox).toBeInTheDocument();
+        await user.click(trackingCheckbox);
+
+        // submit button should be enabled now
+        expect(submitButton).toBeEnabled();
+
+        // mock api call
+        const scope = nock(BASE_API_URL)
+            .patch('/api/v1/user/me/complete', {
+                organizationName: 'test organization',
+                jobTitle: 'Software Engineer',
+                enableEmailDomainAccess: false,
+                isMarketingOptedIn: false,
+                isTrackingAnonymized: true,
+            })
+            .reply(200);
+
+        // submit form
+        await user.click(submitButton);
+
+        // wait for api call to be made
+        scope.done();
+        await waitFor(() => expect(scope.isDone()).toBe(true));
```

Comment:
would it be possible to check if the dialog isn't open anymore?

Reply:
good idea. I might do that in a separate PR tho.